mcp/protocol.go

```diff
@@ -658,6 +658,21 @@ type ProgressNotificationParams struct {
 
 func (*ProgressNotificationParams) isParams() {}
 
+// Icon provides visual identifiers for their resources, tools, prompts, and implementations
+// See [/specification/draft/basic/index#icons] for notes on icons
+
+// TODO(iamsurajbobade): update specification url from draft.
+type Icon struct {
+	// Source is A URI pointing to the icon resource (required). This can be:
+	// - An HTTP/HTTPS URL pointing to an image file
+	// - A data URI with base64-encoded image data
+	Source string `json:"src"`
+	// Optional MIME type if the server's type is missing or generic
+	MIMEType string `json:"mimeType,omitempty"`
+	// Optional size specification (e.g., ["48x48"], ["any"] for scalable formats like SVG, or ["48x48", "96x96"] for multiple sizes)
+	Sizes []string `json:"sizes,omitempty"`
+}
```

Comment:
There is also theme: https://github.com/modelcontextprotocol/modelcontextprotocol/blob/main/schema/draft/schema.ts#L355